importlib_resources/_common.py

```diff
@@ -93,13 +93,22 @@ def _tempfile(
             pass
 
 
+def _temp_file(path):
+    return _tempfile(path.read_bytes, suffix=path.name)
+
+
 @functools.singledispatch
 def as_file(path):
     """
     Given a Traversable object, return that object as a
     path on the local file system in a context manager.
     """
-    return _tempfile(path.read_bytes, suffix=path.name)
+    try:
+        is_dir = path.is_dir()
+    except FileNotFoundError:
+        is_dir = False
```

Comment:
I'm not happy with this behavior. I'd like to extract this behavior and document why it needs to exist.

Reply:
Fixed in bac6e8e3e6.